bazel/rules/cc/defs.bzl

```diff
@@ -6,7 +6,14 @@ def _copts(std):
             "/std:" + std,
             "/Zc:preprocessor",
         ],
-        "//conditions:default": ["-std=" + std],
+        "//conditions:default": [
+            "-std=" + std,
+            "-Werror",
+            # TODO(nils): Clean up type punning violations;
+            # this is easy to do once we get to c++20 because then we
+            # can use std::bit_cast
```

Comment:
Yeah, basically I'm not sure we can get away with not doing type punning, since we're writing such low-level stuff. The Linux kernel also requires type punning, so I'm OK with dropping this warning. Although this really only applies to the guest-side code. I've actually already configured this for the RISC-V toolchain here:
https://github.com/risc0/risc0/blob/main/bazel/toolchain/risc0/gcc_toolchain.bzl#L34
https://github.com/risc0/risc0/blob/main/bazel/toolchain/risc0/gcc_toolchain.bzl#L44

So for the host-side, I could see an argument for removing all type punning since I don't know that it's strictly necessary.

Anyways, this looks good for now.

Reply:
How about we use -fno-strict-aliasing then instead of just disabling the warning?  It seems less error prone.

Reply:
Well on the host-side there are probably important optimizations that can be enabled with strict aliasing. I suppose if we had perf data to support it then that'd be OK.